mfhdf/test/texternal.c

```diff
@@ -842,9 +844,104 @@ test_change_extdir(void)
 
     /* Return the number of errors that's been kept track of so far */
     return num_errs;
-
 } /* test_change_extdir */
 
+/********************************************************************
+   Name: test_HDFFR_1609() - tests writing after creating external file
+                             and without specifying external dir
+   Description:
+    The main contents include:
+    - HXsetdir to a location other than the directory where the external file
+      is going to be created
+    - Create a new hdf file and a new dataset
+    - Set create external directory to the source test directory
+    - Create the external file by calling SDsetexternalfile
+    - Write to the dataset with external data, should succeed (this failed before the fix)
+    - Remove the data file and the external file
+
+   Return value:
+    The number of errors occurred in this routine.
+
+*********************************************************************/
+static int
+test_HDFFR_1609(void)
+{
+    int32 sd_id;
+    int32 sds_id;
+    float sds_data[] = {0.1, 2.3, 4.5, 6.7, 8.9};
+    float sds_out[5];
+    int32 start = 0, stride = 1, edge;
+    int32 dimsize[RANK];
+    char  dir_name[MAX_PATH_LEN]; /* directory from srcdir */
+    char *temp_dir    = NULL;     /* temp dir to create the external file in */
+    int   command_ret = 0;        /* retvalue from system commands */
+    intn  status      = 0;
+    intn  num_errs    = 0; /* number of errors in compression test so far */
+
+    status = make_sourcepath(dir_name, MAX_PATH_LEN);
+    CHECK(status, FAIL, "make_datafilename");
+
+    /* When srcdir is not available, make up a directory to create the external
+       file to complicate the subsequent reads */
```

Comment:
complicate?

Reply:
It means to say that instead of simply looking for the file in the current directory, the test tries to look elsewhere.  Any suggestion for better words?

Reply:
or "allow"